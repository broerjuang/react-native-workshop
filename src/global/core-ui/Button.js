//@flow
import React from 'react';
import {Button as ButtonReactElement} from 'react-native-elements';
import {DARK_GREY} from '../constants/colors';
type Props = {
  styles?: {
    backgroundColor?: string;
    borderRadius?: number;
    underlayColor?: string;
    fontColor?: string;
    fontFamily?: string;
  };
  title: string;
  onPress: (params?: mixed) => void;
};
const Button = (props: Props) => {
  let {styles = {}, onPress, title, ...otherProps} = props;
  let {
    backgroundColor = DARK_GREY,
    borderRadius = 3,
    underlayColor = 'black',
    fontColor = 'black',
    fontFamily = null,
  } = styles;
  return (
    <ButtonReactElement
      borderRadius={borderRadius}
      backgroundColor={backgroundColor}
      title={title}
      underlayColor={underlayColor}
      onPress={onPress}
      color={fontColor}
      fontFamily={fontFamily}
      {...otherProps}
    />
  );
};

export default Button;
