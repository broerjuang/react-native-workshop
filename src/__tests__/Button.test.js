// @flow
import React from 'react';
import Button from '../general/core-ui/Button';
import renderer from 'react-test-renderer';

describe('LoginScreen', () => {
  it('should render LoginScreen corectly', () => {
    let component = renderer.create(
      <Button title={'button'} onPress={() => console.log('tes')} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
