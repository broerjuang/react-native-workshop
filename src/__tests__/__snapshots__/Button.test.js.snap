// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render LoginScreen corectly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      undefined,
      Object {
        "borderRadius": 3,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#909AA3",
          },
          Object {
            "borderRadius": 3,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "black",
              },
              Object {
                "fontSize": 17.5,
              },
              undefined,
              undefined,
              undefined,
              null,
              undefined,
            ],
          ]
        }
      >
        button
      </Text>
    </View>
  </View>
</View>
`;
